### PR TITLE
Review Fixes

### PR DESCRIPTION
**Pull request by prankulmahajan**
Merge and release csi-common before merging -> https://github.com/IBM/ibm-csi-common/pull/28

### CODE REVIEW

File: pkg/ibmcsidriver/controller.go [hunk 1]
```diff
@@ -43,16 +43,29 @@ type CSIControllerServer struct {
 }
 
 const (
-	// PublishInfoVolumeID ...
-	PublishInfoVolumeID = "volume-id"
```
Comment (prankulmahajan) [inline pkg/ibmcsidriver/controller.go:47]: not used

**Comment by prankulmahajan** [comment]
- Gave `VolumeCapability_AccessMode_SINGLE_NODE_WRITER` (RWO Read Write Once) in driver and created PVC with "RWO" access mode.
- Created 3 replicas deployment with 2 pods created in diff nodes.
- All the pods had volume attached and were able to access it.
- But as per RWO the volume must have been attached to a single node and have given an error if tried to attach to diff node.
- "ReadWriteOnce" is not supported in file.

```
➜  ibm-vpc-file-csi-driver git:(review-fixes) ✗ kubectl get pods -o wide| grep testpodrwm
testpodrwm-69dc6b989c-9zv97   1/1     Running   0          33s   172.17.32.209   10.240.64.6   <none>           <none>
testpodrwm-69dc6b989c-c8l67   1/1     Running   0          33s   172.17.3.36     10.240.64.4   <none>           <none>
testpodrwm-69dc6b989c-tjk2m   1/1     Running   0          33s   172.17.32.208   10.240.64.6   <none>           <none>

➜  ibm-vpc-file-csi-driver git:(review-fixes) ✗ kubectl exec testpodrwm-69dc6b989c-9zv97 -it -- bash
root@testpodrwm-69dc6b989c-9zv97:/# cd myvol/
root@testpodrwm-69dc6b989c-9zv97:/myvol# ls
root@testpodrwm-69dc6b989c-9zv97:/myvol# echo "Hi rwo" > test.txt
root@testpodrwm-69dc6b989c-9zv97:/myvol# df -h | grep myvol
fsf-dal1297b-vs90-stg-fz.adn.networklayer.com:/nxg_s_voll_mz0726_a4246ed5_8261_4863_ade6_3e124855d1d3   10G  320K  9.9G   1% /myvol
root@testpodrwm-69dc6b989c-9zv97:/myvol# exit
exit

➜  ibm-vpc-file-csi-driver git:(review-fixes) ✗ kubectl exec testpodrwm-69dc6b989c-c8l67 -it -- bash
root@testpodrwm-69dc6b989c-c8l67:/# cd myvol/
root@testpodrwm-69dc6b989c-c8l67:/myvol# ls
test.txt
root@testpodrwm-69dc6b989c-c8l67:/myvol# cat test.txt 
Hi rwo
root@testpodrwm-69dc6b989c-c8l67:/myvol# df -h | grep myvol
fsf-dal1297b-vs90-stg-fz.adn.networklayer.com:/nxg_s_voll_mz0726_a4246ed5_8261_4863_ade6_3e124855d1d3   10G  320K  9.9G   1% /myvol
root@testpodrwm-69dc6b989c-c8l67:/myvol# 
```

File: cmd/main.go [hunk 2]
```diff
@@ -36,10 +36,10 @@ import (
 	"github.com/IBM/ibm-csi-common/pkg/metrics"
 	mountManager "github.com/IBM/ibm-csi-common/pkg/mountmanager"
 	"github.com/IBM/ibm-csi-common/pkg/utils"
-	"github.com/IBM/ibm-csi-common/pkg/watcher"
 	csiConfig "github.com/IBM/ibm-vpc-file-csi-driver/config"
 	driver "github.com/IBM/ibm-vpc-file-csi-driver/pkg/ibmcsidriver"
-	cloudProvider "github.com/IBM/ibmcloud-volume-file-vpc/ibmcloudprovider"
+	cloudProvider "github.com/IBM/ibmcloud-volume-file-vpc/pkg/ibmcloudprovider"
+	"github.com/IBM/ibmcloud-volume-file-vpc/pkg/watcher"
```
Comment (sameshai) [inline cmd/main.go:42]: Do we really need watcher ? We can discuss with @arahamad once. If tagging is working we should avoid this. as of now let it be there

Comment (prankulmahajan) [reply cmd/main.go:42]: We can avoid it and let it be used from `ibm-csi-common` but then will get references of `ibmcloud-volume-vpc` in go.mod

Comment (sameshai) [reply cmd/main.go:42]: Ok got it , I am talking about removing watcher usage all together for file. We can discuss.

**Review by sameshai — DISMISSED** [review]
We need to also evaluate the NodeGetInfo Call which will set max volume attach limit. Is it applicable to file.

**Review by sameshai — APPROVED** [review]

**Comment by prankulmahajan** [comment]
Tested custom iops support:

```
➜  ibm-vpc-file-csi-driver git:(review-fixes) ✗ kubectl get pods -n kube-system -o wide | grep vpc-file              
ibm-vpc-file-csi-controller-0                        5/5     Running   0          12s   172.17.59.85    10.240.0.4   <none>           <none>
ibm-vpc-file-csi-node-jxtkp                          4/4     Running   0          11s   172.17.59.79    10.240.0.4   <none>           <none>

➜  ibm-vpc-file-csi-driver git:(review-fixes) ✗ kubectl apply -f examples/kubernetes/validPVC.yaml 
persistentvolumeclaim/custom-iops created
➜  ibm-vpc-file-csi-driver git:(review-fixes) ✗ kubectl get pvc                 
NAME                STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS               AGE
csi-file-pvc-good   Bound    pvc-afb8e576-199a-40d6-9bc8-38e3c1bdd997   10Gi       RWX            ibmc-vpc-file-5iops-tier   61m
custom-iops         Bound    pvc-f551a05d-e5c2-4d0c-b85e-aac778d6b663   10Gi       RWX            ibmc-vpc-file-custom       44s
custom-new          Bound    pvc-4ad68925-e713-43e4-818b-323cb5caa341   10Gi       RWX            ibmc-vpc-file-custom       44m
custom-pvc          Bound    pvc-d626f875-5f08-4632-89ca-1a6bc9241523   10Gi       RWX            ibmc-vpc-file-custom       66m

➜  ibm-vpc-file-csi-driver git:(review-fixes) ✗ ibmcloud is shares
Listing shares in resource group vpc-ca-storage and region us-south under account Alchemy Test Squad Id 1's Account as user Prankul.Mahajan@ibm.com...
ID                                          Name                                       Lifecycle state   Zone         Profile       Size(GB)   Resource group   
r006-96fe2ca9-663a-4939-afd9-14826cfd96de   pvc-f551a05d-e5c2-4d0c-b85e-aac778d6b663   stable            us-south-1   custom-iops   10         vpc-ca-storage   
r006-f51e837d-283c-4bc4-8362-d520f0c4127a   pvc-4ad68925-e713-43e4-818b-323cb5caa341   stable            us-south-1   custom-iops   10         vpc-ca-storage   
r006-51add4d7-13cd-4dd2-86d2-5edea352b3ce   pvc-afb8e576-199a-40d6-9bc8-38e3c1bdd997   stable            us-south-1   tier-5iops    10         vpc-ca-storage   
r006-d62e1a92-b9c3-42da-ac98-08c2a8f5fee0   pvc-d626f875-5f08-4632-89ca-1a6bc9241523   stable            us-south-1   custom-iops   10         vpc-ca-storage   
➜  ibm-vpc-file-csi-driver git:(review-fixes) ✗ ibmcloud is share r006-96fe2ca9-663a-4939-afd9-14826cfd96de
Getting file share r006-96fe2ca9-663a-4939-afd9-14826cfd96de under account Alchemy Test Squad Id 1's Account as user Prankul.Mahajan@ibm.com...
                     
ID                r006-96fe2ca9-663a-4939-afd9-14826cfd96de   
Name              pvc-f551a05d-e5c2-4d0c-b85e-aac778d6b663   
CRN               crn:v1:bluemix:public:is:us-south-1:a/c468d8824937fecd8a0860fe0f379bf9::share:r006-96fe2ca9-663a-4939-afd9-14826cfd96de   
Lifecycle state   stable   
Zone              us-south-1   
Profile           custom-iops   
Size(GB)          10   
IOPS              400   
Encryption        provider_managed   
Targets           ID                                          Name                              VPC ID                                      VPC Name      
                  r006-3f055b55-084a-4bf0-98bf-e356fd2cf248   uncross-campus-finally-riveting   r006-439a12ba-0e2b-481b-b7aa-1a36be438455   prankul-vpc      
                     
Resource group    ID                                 Name      
                  f6ac46dbf4594c23b34070a642b9f676   vpc-ca-storage      
                     
Created           2022-08-11T14:39:20+05:30   
Last sync at      0001-01-01T05:53:28+05:53   
➜  ibm-vpc-file-csi-driver git:(review-fixes) ✗ 
➜  ibm-vpc-file-csi-driver git:(review-fixes) ✗ 
                                                                                                                                     
➜  ibm-vpc-file-csi-driver git:(review-fixes) ✗ 
➜  ibm-vpc-file-csi-driver git:(review-fixes) ✗ 
➜  ibm-vpc-file-csi-driver git:(review-fixes) ✗ 
➜  ibm-vpc-file-csi-driver git:(review-fixes) ✗ kubectl describe sc ibmc-vpc-file-custom                   
Name:            ibmc-vpc-file-custom
IsDefaultClass:  No
Annotations:     kubectl.kubernetes.io/last-applied-configuration={"allowVolumeExpansion":true,"apiVersion":"storage.k8s.io/v1","kind":"StorageClass","metadata":{"annotations":{},"name":"ibmc-vpc-file-custom"},"mountOptions":["hard","nfsvers=4.0","sec=sys"],"parameters":{"billingType":"hourly","classVersion":"1","encrypted":"false","encryptionKey":"","gid":"2021","iops":"400","profile":"custom-iops","resourceGroup":"","tags":"","uid":"2020","zone":""},"provisioner":"vpc.file.csi.ibm.io","reclaimPolicy":"Delete"}

Provisioner:           vpc.file.csi.ibm.io
Parameters:            billingType=hourly,classVersion=1,encrypted=false,encryptionKey=,gid=2021,iops=400,profile=custom-iops,resourceGroup=,tags=,uid=2020,zone=
AllowVolumeExpansion:  True
MountOptions:
  hard
  nfsvers=4.0
  sec=sys
ReclaimPolicy:      Delete
VolumeBindingMode:  Immediate
Events:             <none>

```

File: deploy/kubernetes/driver/kubernetes/overlays/stage/controller-server-images.yaml [hunk 3]
```diff
@@ -12,8 +12,8 @@ spec:
...
         - name: iks-vpc-file-driver
           imagePullPolicy: Always
-          image: icr.io/obs/storage/ibm-vpc-file-csi-driver:v1.0.0-beta1
+          image: icr.io/obs/storage/ibm-vpc-file-csi-driver:v1.0.1-beta
         - name: storage-secret-sidecar
-          image: icr.io/obs/armada-storage-secret:v1.1.4
+          image: icr.io/obs/armada-storage-secret:v1.1.12
         - name: csi-resizer
-          image: quay.io/k8scsi/csi-resizer:v1.2.0
+          image: icr.io/ext/sig-storage/csi-resizer:v1.2.0
```
Comment (ambiknai) [inline deploy/kubernetes/driver/kubernetes/overlays/stage/controller-server-images.yaml:19]: In community I see new versions of resizer and other sidecars. Can we consider upgrading .

File: README.md [hunk 4]
```diff
@@ -9,10 +9,10 @@ The following are the supported orchestration platforms suitable for deployment
 |Red Hat® OpenShift®|4.6|x86|
 |Red Hat® OpenShift®|4.7|x86|
 |Red Hat® OpenShift®|4.8|x86|
-|Kubernetes| 1.19|x86|
-|Kubernetes| 1.20|x86|
+|Red Hat® OpenShift®|4.9|x86|
 |Kubernetes| 1.21|x86|
 |Kubernetes| 1.22|x86|
+|Kubernetes| 1.23|x86|
```
Comment (ambiknai) [inline README.md:16]: can we verify in 1.24 as well

**Review by ambiknai — APPROVED** [review]
Review comments are good to be considered as a follow up PR also if applicable. 
This code change looks good to merge.

**Comment by prankulmahajan** [comment]
Hi @ambiknai Will take up the comments in the upcoming PR

**Pull request merged**